### ISSUE TITLE
Add a context menu in the Code Explorer

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#231]
Let's add a right-click context menu for the Code Explorer tree nodes.

- [x] **Navigate**, to activate a code pane for the selected source file - if module-level, selection should be (1,1,1,1).
- [x] **Rename...**, to rename node / project / module / member, and update all references.in project.
- [ ] **Safe delete**, to remove a component without breaking the code.
- [x] Add > **Test Module**, to add a new test module.
- [x] Add > **Class**, to add a new class module.
- [x] Add > **Module**, to add a new standard module.
- [x] Add > **Form**, to add a new form.
- [x] Source Control > **Commit**, to save changes to local repository.
- [x] Source Control > **Undo**, to undo *uncommitted* changes to a file.

**Comment by ckuhn203** [comment]
I like this a lot. Good call.

**Comment by ckuhn203** [comment]
Random thought. This is really a *project* explorer at this point. And since files will be stored in an external directory anyway, why not leverage that somehow to allow for adding directories to make it easier to organize a project. I think we can manage this by putting a `.\Rubberduck\Project\` directory in the app data folder. Each VBProject gets it's own xml file that describes the directory tree. Then (as VBA devs) we could actually organize projects in some sensible way.  Just a thought.

**Comment by retailcoder** [comment]
This is a very interesting idea, but there are a number of caveats: VBA has no namespaces, so file names / modules shouldn't be allowed to have the same name despite being in a different folder.

We could avoid all this hassle with *virtual* folders instead - keep all physical files in one folder, and have some `vbproject1.rubberduck` file to hold the "folder" names for each component; the tree listener only needs that mapping to generate the proper folder nodes.

**Comment by ckuhn203** [comment]
I don't think virtual folders cut it. I would want the same folder structure that's in my IDE to show up in my Git repo. It's certainly an option though.

**Comment by retailcoder** [comment]
I really like the folders idea, but I think it would be [much] simpler to implement them as *virtual* ones, leaving all code files in the same folder and thus having the OS enforce the uniqueness of file names... but that discussion/feature deserves its own issue number I think ;)

**Comment by ckuhn203** [comment]
Is this more or less completed by #292? Perhaps we need to mark this with the source control label and finish it off in v1.3?

**Comment by retailcoder** [comment]
I wouldn't tag with *Source Control*, since it's really *Code Explorer* UI work. But yeah all that's left to implement needs features planned for 1.3, so moving this one to 1.3 is the right thing to do :)

**Comment by retailcoder** [comment]
This will be much easier to implement after the source control functionality is refactored into commands.

**Comment by Hosch250** [comment]
There is now no context menu for the code explorer.

**Comment by retailcoder** [comment]
@Hosch250 right. But now that we have commands for source control, we can fully implement it!

**Comment by Hosch250** [comment]
Safe Delete won't be done until 2.1 at least.  This can be closed once #1467 is merged.
